### PR TITLE
added the .gitattributes file to test composer packaging and distribution

### PR DESCRIPTION
**Pull request by mikelking**
This is directly related to the root repo's issue [649](https://github.com/wp-graphql/wp-graphql/issues/649)

As per the recommendation I am testing this in accordance with https://madewithlove.be/gitattributes/

### CODE REVIEW

**Comment by mikelking** [comment]
As far as I can tell the intended use of gitattributes does not state anything along the lines mentioned in the previous site. According to the git manual it is really a way to describe text conversions and conventions necessary for DIFF. So far it have not added the exclusion desired.

**Comment by mikelking** [comment]
Ok I have conducted considerable testing with my personal project I have conducted https://github.com/mikelking/bacon and not sure exactly how but successfully used .gitattribute to change the composer artifacts. This construct is a bit of a pain in that it needs to be merged into an install referenceable version for testing with composer.

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/tests export-ignore
+/.gitattributes export-ignore
+/.gitignore export-ignore
+/phpcs.ruleset.xml  export-ignore
+/phpunit.xml.dist export-ignore
+/ISSUE_TEMPLATE.md export-ignore
+/PULL_REQUEST_TEMPLATE.md export-ignore
+#/changelog.md export-ignore
+#/readme.md export-ignore
+/vendor export-ignore
```
Comment (jasonbahl) [inline .gitattributes]: so, at the moment, if you were to download the Zip file from the Github releases, the `vendor` directory _is_ included, as we want folks to be able to download the Zip, install in WordPress and be up and running. 

So we still need to include the `vendor` directory, at least for the time being

**Comment by mikelking** [comment]
@jasonbahl ,

I made the change by commenting our the vendor path from the exclusion. I also left the readme & change log as I believe those are beneficial to end users. Many examples on the web leave those out.

**Comment by jasonbahl** [comment]
@CodeProKid any thoughts you want to add to this?

**Comment by CodeProKid** [comment]
I think there are still some additional things that can get ignored.

- bin/
- img/
- CODE_OF_CONDUCT.md
- CONTRIBUTING.md

**Pull request merged**